packages/cli/src/commands/init/init.js

```diff
@@ -104,6 +107,87 @@ async function createFromTemplate({
   }
 }
 
+async function installPods(loader: typeof Ora) {
+  process.chdir('ios');
+
+  const hasPods = await fs.pathExists('Podfile');
+
+  if (!hasPods) {
+    return;
+  }
+
+  try {
+    await commandExists('pod');
+  } catch (err) {
+    loader.succeed();
+
+    const {shouldInstallCocoaPods} = await inquirer.prompt([
+      {
+        type: 'confirm',
+        name: 'shouldInstallCocoaPods',
+        message: 'CocoaPods is not installed, do you want to install it?',
+      },
+    ]);
+
+    if (shouldInstallCocoaPods) {
+      try {
+        // First attempt to install `cocoapods`
+        await execa('gem', ['install', 'cocoapods'], {
+          stdio: 'pipe',
+        });
+      } catch (err) {
+        try {
+          // If that doesn't work then try with sudo
+          await execa('sudo', ['gem', 'install', 'cocoapods'], {
+            stdio: 'pipe',
+          });
+        } catch (err) {
+          throw new CLIError(
+            'Error occurred while trying to install CocoaPods, please run this command again.',
+            err,
+          );
+        }
+      }
+
+      loader.start('Installing pods');
```

Comment:
this should be moved right before `await execa('pod', ['install'])` is called

Reply:
This is the reason it's not after the condition: https://github.com/react-native-community/cli/pull/373#discussion_r282442627.

Reply:
A comment with the explanation would be nice, because it seems unintuitive

Reply:
I'll add it.